dev/bots/cirrus_setup.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+set -e
+
+# This script is only meant to be run by the Cirrus CI system, not locally.
+# It must be run from the root of the Flutter repo.
+
+# Collects log output in a tmpfile, but only prints it if the command fails.
+function log_on_fail() {
+  local COMMAND="$@"
+  local TMPDIR="$(mktemp -d)"
+  local TMPFILE="$TMPDIR/command.log"
+  local EXIT=0
+  if ("$@" > "$TMPFILE" 2>&1); then
+    echo "'$COMMAND' succeeded."
+  else
+    EXIT=$?
+    cat "$TMPFILE" 1>&2
+    echo "FAIL: '$COMMAND' exited with code $EXIT" 1>&2
+  fi
+  rm -rf "$TMPDIR"
+  return "$EXIT"
+}
+
+echo "Flutter SDK directory is: $PWD"
+
+# Run flutter to download dependencies and precompile things, and to disable
+# analytics on the bots.
+echo "Downloading build dependencies and pre-compiling Flutter snapshot"
+log_on_fail ./bin/flutter config --no-analytics
+
+# Run doctor, to print it to the log for debugging purposes.
+./bin/flutter doctor -v
+
+# Accept licenses.
```

Comment:
Because I'm not setting up Flutter itself in the Docker image. I can't do that (at least not in the same location), because the cloning of the repo will fail during builds then.

I suppose I could make the directory and touch the file to silence the warning, however.

Reply:
I wonder if setup the Flutter in the Docker image would further accelerate the process. We can also update the image every day or every week to make the cache fresh so it will keep fast.